layer/support/trace_event_logging.cc

```diff
@@ -38,12 +44,22 @@ std::string ValueToJsonString(const std::vector<int64_t> &values) {
   return json_str.str();
 }
 
-std::string ValueToJsonString(Duration value) {
-  return std::to_string(value.ToMilliseconds());
+std::string_view ValueToJsonString(std::array<char, kBufferSize> &buffer,
```

Comment:
Why create a local `std::string`, use it as the buffer, and return at the end? You can get the pointer to the underlying storage by calling `.data()`

Reply:
Sure. Since the size of the underlying buffer can vary, using a dynamic allocation approach seems more reasonable.